src/dice.class.php

```diff
@@ -46,22 +46,27 @@ function connectDatabase() {
 	 */
 	function requireMailsAreRegistered(array $emails) {
 		$this->connectDatabase();
-		$emails_string = "'" . implode("', '", $emails) . "'";
-		$sql = "SELECT registered_email FROM dice_emails WHERE registered_email IN ($emails_string)";
-		$result = $this->dbconn->query($sql) or exit("fatal error: data connection lost @requireMailsAreRegistered!");
-		$registered_mails = $result->fetch_array();
-		if ($result->num_rows === count($emails)) {
-			return;	// all emails are registered
+		$emails = array_unique($emails);
+		$emailCount = count($emails);
+		$placeholder = implode(", ", array_fill(0, $emailCount, "?"));
+		$sql = "SELECT COUNT(*), registered_email FROM dice_emails WHERE registered_email IN ($placeholder) GROUP BY registered_email";
+		$statement = $this->dbconn->prepare($sql);
+		$statement->bind_param(str_repeat("s", $emailCount), ...$emails);
```

Comment:
I'm not sure the spread operator works here.  From my testing with two emails, it's as if the second email is being bound twice such that the `IN` clause is effectively `IN ('email2', 'email2')` instead of `IN ('email1', 'email2')`.

[This comment](http://php.net/manual/en/mysqli-stmt.bind-param.php#104073) in the docs for `bind_param()` seems to indicate that there may be issues when passing array elements due to how references are tracked.

Reply:
Ignore my last comment.  I verified via the MySQL query log that the final prepared statement is correct.  I'll keep looking to determine why my query result contains two copies of the second email.

Reply:
Derp.  It was my fault.  I tried to get too cute with the `fetch_array()` function.